src/Network/Pusher/Data.hs

```diff
@@ -59,26 +64,48 @@ type AppSecret = B.ByteString
 
 -- |All the required configuration needed to interact with the API.
 data Pusher = Pusher
-  { pusherHost :: T.Text
-  , pusherPath :: T.Text
-  , pusherCredentials :: Credentials
+  { pusherHost              :: T.Text
+  , pusherPath              :: T.Text
+  , pusherCredentials       :: Credentials
   , pusherConnectionManager :: Manager
   }
 
 -- |The credentials for the current app.
 data Credentials = Credentials
-  { credentialsAppID :: AppID
-  , credentialsAppKey :: AppKey
+  { credentialsAppID     :: AppID
+  , credentialsAppKey    :: AppKey
   , credentialsAppSecret :: AppSecret
+  , credentialsCluster   :: Maybe Cluster
   }
 
 instance A.FromJSON Credentials where
   parseJSON (A.Object v) = Credentials
     <$> v .: "app-id"
     <*> (encodeUtf8 <$> v .: "app-key")
     <*> (encodeUtf8 <$> v .: "app-secret")
+    <*> v.:? "app-cluster"
   parseJSON v2 = failExpectObj v2
 
+-- | The cluster the current app resides on. Common clusters include: mt1,eu,ap1,ap2
+newtype Cluster = Cluster {clusterName :: T.Text}
+
+clusterMt1, clusterEu, clusterAp1, clusterAp2 :: Cluster
+clusterMt1 = Cluster "mt1"
+clusterEu  = Cluster "eu"
+clusterAp1 = Cluster "ap1"
+clusterAp2 = Cluster "ap2"
```

Comment:
Hmm yes that is a good point. Leave it as is. It really is an arbitrary string; those constants are just for connivence.